src/libraries/Common/src/System/Net/Mail/MailAddressParser.cs

```diff
@@ -296,27 +423,55 @@ private static string ParseDisplayName(string data, ref int index, bool expectMu
                 int startingIndex = index;
 
                 // Read until the dividing comma or the end of the line.
-                index = QuotedStringFormatReader.ReadReverseUnQuoted(data, index, true, expectMultipleAddresses);
+                if (!QuotedStringFormatReader.TryReadReverseUnQuoted(data, index, true, expectMultipleAddresses, out index, throwExceptionIfFail))
+                {
+                    Debug.Assert(!throwExceptionIfFail);
+                    return false;
+                }
 
                 Debug.Assert(index < 0 || data[index] == MailBnfHelper.Comma, "Mis-aligned index: " + index);
 
                 // Do not include the Comma (if any), and because there were no bounding quotes,
                 // trim extra whitespace.
                 displayName = data.SubstringTrim(index + 1, startingIndex - index);
             }
-            return NormalizeOrThrow(displayName);
+
+            if (!TryNormalizeOrThrow(displayName, out displayName, throwExceptionIfFail))
+            {
+                Debug.Assert(!throwExceptionIfFail);
+                return false;
+            }
+
+            return true;
         }
 
-        internal static string NormalizeOrThrow(string input)
+        internal static bool TryNormalizeOrThrow(string input, out string normalizedString, bool throwExceptionIfFail)
         {
             try
             {
-                return input.Normalize(Text.NormalizationForm.FormC);
+                normalizedString = input.Normalize(Text.NormalizationForm.FormC);
+                return true;
             }
             catch (ArgumentException e)
             {
-                throw new FormatException(SR.MailAddressInvalidFormat, e);
+                if (throwExceptionIfFail)
+                {
+                    throw new FormatException(SR.MailAddressInvalidFormat, e);
+                }
+                else
+                {
+                    normalizedString = default;
+                    return false;
+                }
             }
         }
     }
+
+    internal class ParseAddressInfo
+    {
+        public string DisplayName { get; private set; }
+        public string User { get; private set; }
+        public string Host { get; private set; }
+        public ParseAddressInfo(string displayName, string userName, string domain) => (DisplayName, User, Host) = (displayName, userName, domain);
+    }
```

Comment:
Why is this class needed?

Reply:
Do you prefer all single out param?It's only to have more compact code.

Reply:
I don't have a strong preference.  Was it individual out params before? Should this be a struct?

Reply:
This is new out before from parsing was returned `MailAddress` https://github.com/dotnet/runtime/blob/master/src/libraries/Common/src/System/Net/Mail/MailAddressParser.cs#L128
Yup could be a struct

Reply:
I tried also with a tuple but I don't like it.

Reply:
Why wasn't the transformation to just make the MailAddress be the out?

Reply:
Because the returned instance is not a "complete" `MailAddress` and is used like DTO(_displayNameEncoding  is not parsed)

https://github.com/dotnet/runtime/blob/master/src/libraries/System.Net.Mail/src/System/Net/Mail/MailAddress.cs#L92

So I thought to cleanup a bit and separate parsing outcome from who use it.

Reply:
I see.  Ok.  Let's go with the struct.